esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -0,0 +1,204 @@
+        formatOUname(respGroupsBaseDN, req.getEnvName(), req.getGroupsMode(), sharedGroupsBaseDN.str(), req.getCustomGroupsBaseDN(), "ou=groups");
+        formatOUname(respUsersBaseDN, req.getEnvName(), req.getUsersMode(), sharedUsersBaseDN.str(), req.getCustomUsersBaseDN(), "ou=users");
+        formatOUname(respResourcesBaseDN, req.getEnvName(), req.getResourcesMode(), sharedResourcesBaseDN.str(), req.getCustomResourcesBaseDN(), "ou=smc,ou=espservices");
+        formatOUname(respWorkunitsBaseDN, req.getEnvName(), req.getWorkunitsMode(), sharedWorkunitsBaseDN.str(), req.getCustomWorkunitsBaseDN(), "ou=workunits");
+
+        // Create HPCC Admin Username
+
+        VStringBuffer respAdminUser("admin_%s", req.getEnvName());
+
+
+        //Create LDAP resources OU Hierarchy
+
+        if (req.getCreateLDAPEnvironment())
+        {
+            VStringBuffer description("%s, created by %s", req.getEnvDescription(), req.getEnvOwnerName());
+            try { secmgr->createLdapBasedn(nullptr, respFilesBaseDN.str(), PT_ADMINISTRATORS_ONLY, description.str()); }
+            catch(...) {}
+            try { secmgr->createLdapBasedn(nullptr, respGroupsBaseDN.str(), PT_ADMINISTRATORS_ONLY, description.str()); }
+            catch(...) {}
+            try { secmgr->createLdapBasedn(nullptr, respUsersBaseDN.str(), PT_ADMINISTRATORS_ONLY, description.str()); }
+            catch(...) {}
+            try { secmgr->createLdapBasedn(nullptr, respResourcesBaseDN.str(), PT_ADMINISTRATORS_ONLY, description.str()); }
+            catch(...) {}
+            try { secmgr->createLdapBasedn(nullptr, respWorkunitsBaseDN.str(), PT_ADMINISTRATORS_ONLY, description.str()); }
+            catch(...) {}
+
+            // NOTE!  HPCC Admin group and HPCC Admin user will be
+            // created when ECLWatch is started in Admin mode
+        }
+        
+        //----------------------------------
+        // Set response
+        //----------------------------------
+        resp.setHPCCAdminUsername(respAdminUser.str());
+
+        VStringBuffer ldapHelm("\n"
+                               "  ldap:\n"
+                               "    adminGroupName: %s\n"
+                               "    hpccAdminSecretKey: TODO!  Create Kubernetes Secret for %s\n"
+                               "    filesBasedn: %s\n"
+                               "    groupsBasedn: %s\n"
+                               "    usersBasedn: %s\n"
+                               "    resourcesBasedn: %s\n"
+                               "    workunitsBasedn: %s\n",
+                               adminGroupName.str(), respAdminUser.str(), respFilesBaseDN.str(), respGroupsBaseDN.str(),
+                               respUsersBaseDN.str(), respResourcesBaseDN.str(), respWorkunitsBaseDN.str());
+        resp.setLDAPHelm(ldapHelm.str());
+        resp.setRetcode(0);
```

Comment:
You can always add the retcode field later when you are actually going to set it to something other than 0.

Reply:
Retcode remove from both methods